spec/classes/cluster_roles_spec.rb

```diff
@@ -60,6 +61,7 @@
           'controller_address' => '172.17.10.101',  
           'node_label' => 'foo',   
           'container_runtime' => 'docker',  
+          'ignore_preflight_errors' => '',  
```

Comment:
For me it can be setted to undef like some other paramters but it does works with it too which is weird.  If you check my first post, you'll see that being empty is not an issue ` --discovery-token-ca-cert-hash 'sha256:TOKEN_SHA256' --ignore-preflight-errors ''` so it's the specs are messing up ...

Reply:
It's not `kubeadm` complaining about it being a `String`, it's Puppet. Because the parameter has the type `Optional[Array]` it can only be passed into the Puppet manifest as `undef` or an `Array`.

Reply:
I know it's not kubeadm, as said previously it's the spec, I'll fix it later ^^